tiatoolbox/wsicore/wsireader.py

```diff
@@ -2809,15 +2818,22 @@ def __init__(
             mpp=mpp,
             power=power,
         )
-        if mode.lower() not in ["rgb", "bool"]:
+        if mode.lower() not in ["rgb", "bool", "feature"]:
             msg = "Invalid mode."
             raise ValueError(msg)
-        self.mode = mode.lower()
+
         if isinstance(input_img, np.ndarray):
             self.img = input_img
         else:
             self.img = utils.imread(self.input_path)
 
+        if mode != "bool" and (
+            self.img.ndim == 2 or self.img.shape[2] not in [3, 4]  # noqa: PLR2004
+        ):
+            mode = "feature"
```

Comment:
Logging would be better option here as background composite would fail in case the input is not rgb or rgba. Also, node should be bool for a boolean input.